app/containers/ReactionsList.tsx

```diff
@@ -103,35 +135,102 @@ const ReactionsTabBar = ({ tabs, activeTab, goToPage, baseUrl, getCustomEmoji, w
 
 const UsersList = ({ tabLabel }: { tabLabel: IReaction }) => {
 	const { colors } = useTheme();
-	const { emoji, usernames } = tabLabel;
+	const { emoji, usernames, names } = tabLabel;
+	const users =
+		names?.length > 0
+			? usernames.map((username, index) => ({ username, name: names[index] }))
+			: usernames.map(username => ({ username, name: '' }));
 	return (
 		<FlatList
-			data={usernames}
+			data={users}
 			ListHeaderComponent={() => (
 				<View style={styles.emojiName}>
 					<Text style={{ color: colors.auxiliaryTintColor }}>{emoji}</Text>
 				</View>
 			)}
 			renderItem={({ item }) => (
 				<View style={styles.userItemContainer}>
-					<Avatar text={item} size={36} />
-					<View style={styles.usernameContainer}>
-						<Text style={[styles.usernameText, { color: colors.titleText }]}>{item}</Text>
+					<Avatar text={item.username} size={36} />
+					<View style={styles.textContainer}>
+						{item.name ? (
```

Comment:
It's not considering `useRealName` and it's prioritizing `item.name`.
If `false`, the workspace admin wants to render usernames everywhere.

Reply:
But `item.name` is present only when `useRealName` is enabled on the server. Names array would be empty otherwise and it will render just the username. I can change the logic if you want.